subprojects/parseq/src/test/java/com/linkedin/parseq/ParSeqUnitTestHelperTest.java

```diff
@@ -14,4 +17,23 @@
         assertThrows(AssertionError.class, () -> ParSeqUnitTestHelper.assertEquals("x", null));
         assertThrows(AssertionError.class, () -> ParSeqUnitTestHelper.assertEquals("x", "y"));
     }
+
+  @Test(description = "Proves that runAndWaitForPlanToCompleteException verifies the expected exception is thrown "
+    + "while also waiting for entire plan to complete")
+  public void testRunAndWaitForPlanToCompleteException() throws Exception {
+    ParSeqUnitTestHelper parSeqUnitTestHelper = new ParSeqUnitTestHelper();
+    parSeqUnitTestHelper.setUp();
+
+    Task<String> delayedSideEffect = parSeqUnitTestHelper.delayedValue("delayed", 1, TimeUnit.SECONDS);
+    parSeqUnitTestHelper.runAndWaitForPlanToCompleteException(
+      Task.par(
+        Task.value("a").withSideEffect(a -> delayedSideEffect),
+        parSeqUnitTestHelper.delayedFailure(new Exception(), 100, TimeUnit.MILLISECONDS)
```

Comment:
Not really. Actually such a test is not possible because if only sideEffect throws an exception, then the plan will complete with success and hence the test would fail if runAndWaitExceptionOnPlanCompletion is used. 
If you want I can write such a test to prove that Exception is not thrown by the plan and sideEffect is also complete.

Reply:
Got you.